Libraries/PermissionsAndroid/PermissionsAndroid.js

```diff
@@ -119,36 +113,25 @@ class PermissionsAndroid {
    * and then shows the system permission dialog
    */
   requestPermission(permission: string, rationale?: Rationale) : Promise<boolean> {
-    return new Promise((resolve, reject) => {
-      const requestPermission = () => {
-        AndroidPermissions.requestPermission(
-          permission,
-          function(perm: string, result: boolean) { resolve(result); },
-          function(error: string) { reject(error); },
-        );
-      };
+    return new Promise(async (resolve, reject) => {
+      const requestPermission = () => resolve(Permissions.requestPermission(permission));
 
       if (rationale) {
-        AndroidPermissions.shouldShowRequestPermissionRationale(
-          permission,
-          function(perm: string, shouldShowRationale: boolean) {
-            if (shouldShowRationale) {
-              DialogManagerAndroid.showAlert(
-                rationale,
-                () => {
-                  DialogManagerAndroid.showAlert({
```

Comment:
I think it's better left to the developer about what happens when requesting a permission fails.